localstack-core/localstack/config.py

```diff
@@ -1032,6 +1032,13 @@ def populate_edge_configuration(
 # Whether to really publish to GCM while using SNS Platform Application (needs credentials)
 LEGACY_SNS_GCM_PUBLISHING = is_env_true("LEGACY_SNS_GCM_PUBLISHING")
 
+# Whether to enable the Next Gen APIGW invocation logic (handler chain)
+# TODO: remove this before merge!
+if not os.environ.get("PROVIDER_OVERRIDE_APIGATEWAY"):
+    os.environ["PROVIDER_OVERRIDE_APIGATEWAY"] = "next_gen"
+
+APIGW_NEXT_GEN_PROVIDER = os.environ.get("PROVIDER_OVERRIDE_APIGATEWAY", "") == "next_gen"
```

Comment:
nit: I think the term `next_gen` is not very future proof (next one would be "next next gen"? 😛), but honestly I'm struggling a bit to find a better name as well (as `v2` would be highly confusing 😛).

Reply:
Yeah, @cloutierMat had the exact (I mean, exactly exact) comment about the "next gen" value. I'd like to find something better too, maybe something related to the handler chain? but then also "leak" implementation detail through the feature flag.  I've been lazy and took the work stream name 😅

Reply:
Could we instead not name this one and already name the current one "LEGACY"? and turn that on by default. then remove the "LEGACY" default when we are ready?

Reply:
That might a bit tricky, because we would need to already override the default, also change it in -ext to override the default and then we'd lose the ability to fall back to the current implementation for -ext.

For now it's still not set in stone so we might be able to still find a better name! 😄

Reply:
I figured it might not be ideal... Let's keep the gear rolling! :gear:

Reply:
Yeah, I was also lacking some creativity for a better name :D This was really just a nitpick ;)